src/sentry/migrations/0182_update_user_misery_on_saved_queries.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 1.11.29 on 2021-03-26 14:43
+import re
+
+from django.db import migrations
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+USER_MISERY_REGEX = r"^user_misery\((\d+)\)"
+
+
+def update_user_misery_column_on_saved_queries(apps, schema_editor):
+    DiscoverSavedQuery = apps.get_model("sentry", "DiscoverSavedQuery")
+
+    for saved_query in RangeQuerySetWrapperWithProgressBar(DiscoverSavedQuery.objects.all()):
+        query = saved_query.query
+        if "fields" not in query:
+            continue
+
+        fields = query["fields"]
+        for i, field in enumerate(fields):
+            match = re.match(USER_MISERY_REGEX, field)
+            if match and "count_unique(user)" not in fields:
```

Comment:
Is the assumption here is that there will only ever be at most one matching field?

Reply:
+1, since we know none of the saved queries currently have >1 misery field i think this might be fine. But we should at least move the query save out of the field loop. eg. this is valid today:
![image](https://user-images.githubusercontent.com/4205004/112694470-cfd45f00-8e58-11eb-9dd6-2e35d21af8cf.png)